### ISSUE TITLE
What kinds of enhancements is this project open to?

### ISSUE THREAD

**Issue by David-Guillot** [wsvincent/djangox#139]
Hi @wsvincent ,

Thanks you for this initiative, I'm sure it helps many people :+1: 

I have been thinking of doing something similar but I think what the community needs is not a multitude of initiatives but a big visible/popular project like this one, so I'd like to know what the contributing strategy is, based on what is written in [the current contribution guidelines](https://github.com/wsvincent/djangox/blob/094677b7347ee2b833e7ce010606da989df09f3e/CONTRIBUTING.md).

I see some ways it could be improved, and I would be willing to open PRs in these areas:
- Dependencies management: this summer, [uv](https://astral.sh/blog/uv-unified-python-packaging) changed everything ; it could be an opportunity to switch to a `pyproject.toml` file, with un-pinned high-level dependencies, and simplify the installation process, what do you think?
- Settings and environment variables: I think that as soon as a Django project gets bigger than a simple blog, settings are better split. And as soon as you want to deploy to production, development settings need to live separately. And of course, environment variables can be integrated through `.env` files.
- Last but not least: frontend stuff. Do you think this project would benefit a `package.json` that would store high-level un-pinned dependencies, with an install/build/bundle script, leveraging, say, [esbuild](https://esbuild.github.io/)?

Let me know what do you think of all this :wink:

**Comment by wsvincent** [comment]
Hi @David-Guillot! Thanks for reaching out.

You are right that `uv` is a game changer. I will admit I haven't dived into it much myself, but if you wanted to add an option to manage dependencies with it, I'd love to review that and probably add it to the existing ones.

Re settings/environment variables, I'm personally a fan of environment variables and ONE SETTINGS FILE TO RULE THEM ALL. I know not everyone feels this way. If I expanded this project to include deployment, this is where I'd start, but I feel the project benefits from not bloating out into all that. If you wanted to do an open source version with it, go for it! I'm sure people would be interested.

Re frontend stuff... this also feels beyond scope for what I have in mind. But it would also be of interest to people I'm sure.

So tl;dr, I'd love if you were up for adding `uv` support. If you are passionate about all three, it's probably simplest if you fork or create your own version for that!

**Comment by David-Guillot** [comment]
Hi @wsvincent ,

Thanks for sharing your clear vision for this project template.

As you know I released [something of my own](https://codeberg.org/David-Guillot/django-project-template) today, since I have quite strong opinions both on the settings/envvars business and on the front-end area, and since I realize our templates don't have the same target audience.

But I will most likely open a PR in the next few weeks for your template to handle dependencies with `uv` :wink: 

Closing this issue, opening one dedicated to `uv`.
